keepalived/vrrp/vrrp_print.c

```diff
@@ -418,3 +421,182 @@ vrrp_print_stats(void)
 	}
 	fclose(file);
 }
+
+#ifdef _JSON_
+void
+vrrp_print_json(void)
+{
+	FILE *file;
+	file = fopen ("/tmp/keepalived.json","w");
+	if (!file) {
+		log_message(LOG_INFO, "Can't open /tmp/keepalived.json (%d: %s)",
+			errno, strerror(errno));
+		return;
+	}
+
+	list l = vrrp_data->vrrp;
+	if LIST_ISEMPTY(vrrp_data->vrrp) {
+		log_message(LOG_INFO, "vrrp_data->vrrp is empty");
+		return;
+	}
+
+	element e, f;
+	vrrp_t *vrrp;
+	struct json_object *array, *vips;
```

Comment:
The above three lines should be at the start of the function, or at the start of the loop, depending on how you resolve the next comment.